client/lock.go

```diff
@@ -242,18 +242,14 @@ func (c *lockClient) waitDelete(pfx string, myRev int64) {
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 		watcher, _ := c.watchClient.Watch(ctx, lastKey)
+	outer:
 		for r := range watcher {
-			f := false
 			for _, e := range r.Events {
 				if e.Type == xlineapi.Event_DELETE {
-					f = true
-					break
+					cancel()
```

Comment:
There's already a defer on the cancel, so there's no need to call it explicitly

Reply:
Done.